packages/protobuf/package.json

```diff
@@ -25,9 +25,9 @@
   "main": "./dist/cjs/index.js",
   "exports": {
     ".": {
-      "module": "./dist/esm/index.js",
+      "import": "./dist/proxy/index.js",
       "require": "./dist/cjs/index.js",
-      "import": "./dist/proxy/index.js"
+      "module": "./dist/esm/index.js"
```

Comment:
This will cause webpack, rollup, wmr ([ref](https://webpack.js.org/guides/package-exports/#conditions)), esbuild ([ref](https://esbuild.github.io/api/#how-conditions-work)) to choose "import" over "module".

I don't think we can do that, because dist/proxy/index.js is an ESM wrapper that re-exports CJS to avoid the dual package hazard. I'm pretty sure it will cause problems for tree shaking, and possibly other issues.

For reference, here is how the current exports were tested: https://github.com/connectrpc/examples-es/pull/1002

I think some other solution is necessary to fix the issue, while maintaining behavior for other consumers. I'm not entirely sure which one yet 🙂

Reply:
I'm pretty sure we can find some better solutions to support dual package hazard and replace proxy with an alternative, yet make a fully standard ESM package with a standard `import` export condition that works natively. (if this is your main concern to resolve) -- Feel free to direct me in discord (pi0), i can share my 50 cents of experience with dealing with ESM/CJS dual exports.